src/pretix/base/models/auth.py

```diff
@@ -353,6 +356,43 @@ def send_security_notice(self, messages, email=None):
         except SendMailException:
             pass  # Already logged
 
+    def send_confirmation_code(self, reason, email=None):
+        from pretix.base.services.mail import mail
+
+        with language(self.locale):
+            if reason == 'email_change':
+                msg = str(_('to confirm changing your email address from {old_email}\nto {new_email}, use the following code:').format(
+                    old_email=self.email, new_email=email,
+                ))
+            else:
+                raise Exception('Invalid confirmation code reason')
+
+        code = "%07d" % random.randint(0, 9999999)
+        cache.set('user_confirmation_code:' + str(self.pk), code + ':' + reason + ':' + str(email), 1800)
+
+        mail(
+            email or self.email,
+            _('pretix confirmation code'),
+            'pretixcontrol/email/confirmation_code.txt',
+            {
+                'user': self,
+                'reason': msg,
+                'code': code,
+            },
+            event=None,
+            user=self,
+            locale=self.locale
+        )
+
+    def check_confirmation_code(self, reason, code):
```

Comment:
todo: this method needs a limit on how many times it can be called on the same generated code, to prevent brute force attacks